### PR TITLE
Time unit parameterization

### PR DESCRIPTION
**Pull request by jaydoane**

### CODE REVIEW

File: src/ets_lru.erl [hunk 1]
```diff
@@ -173,10 +179,11 @@ handle_call({insert, Key, Val}, _From, St) ->
             true = ets:delete(St#st.atimes, ATime),
             true = ets:insert(St#st.atimes, {NewATime, Key});
         [] ->
-            Entry = #entry{key=Key, val=Val, atime=NewATime, ctime=NewATime},
+            NewCTime = element(1, NewATime),
```
Comment (iilyak) [inline src/ets_lru.erl]: Please avoid magic numbers here. Couple of options:
option 1:
```
line 173:
  {NewCTime, _} = NewATime = strict_monotonic_time(St#st.time_unit),
```

option 2:
```
{NewCTime, _} = NewATime,
```

File: src/ets_lru.erl [hunk 2]
```diff
@@ -173,10 +179,11 @@ handle_call({insert, Key, Val}, _From, St) ->
...
+            Entry = #entry{key=Key, val=Val, atime=NewATime, ctime=NewCTime},
             true = ets:insert(St#st.objects, Entry),
             true = ets:insert(St#st.atimes, {NewATime, Key}),
-            true = ets:insert(St#st.ctimes, {NewATime, Key})
+            true = ets:insert(St#st.ctimes, {NewCTime, Key})
```
Comment (iilyak) [inline src/ets_lru.erl]: NewCTime is Erlang's monotonic time. According to documentation:
> consecutive calls to erlang:monotonic_time/0 can produce the same result.

File: src/ets_lru.erl [hunk 3]
```diff
@@ -332,6 +339,8 @@ set_options(St, [{max_size, N} | Rest]) when is_integer(N), N >= 0 ->
     set_options(St#st{max_size=N}, Rest);
 set_options(St, [{max_lifetime, N} | Rest]) when is_integer(N), N >= 0 ->
     set_options(St#st{max_lifetime=N}, Rest);
+set_options(St, [{time_unit, T} | Rest]) when is_atom(T) ->
```
Comment (iilyak) [inline src/ets_lru.erl]: Is there a reason to make it configurable?

File: src/ets_lru.erl (hunk 1 above)
Comment (jaydoane) [reply src/ets_lru.erl]: Yes, I think this was a code smell indicating that `ctime` and `atime` should not have different shapes. I've removed it in latest version.

File: src/ets_lru.erl (hunk 2 above)
Comment (jaydoane) [reply src/ets_lru.erl]: I must have thought for some reason that `ctime` table entries were not required to be unique, and that we could save some memory by not storing that extra integer. However, after looking at [this code](https://github.com/apache/couchdb-ets-lru/blob/master/src/ets_lru.erl#L286) and [also this](https://github.com/apache/couchdb-ets-lru/blob/master/src/ets_lru.erl#L306) it does indeed appear that `ctime` entries are expected to be unique, and so like `atime`, also of type `strict_monotonic_time()`. It's been changed in the latest commit.

File: src/ets_lru.erl (hunk 3 above)
Comment (jaydoane) [reply src/ets_lru.erl]: It requires relatively little overhead, and allows the user to choose something other than millisecond resolution if their use case requires it. However if you don't like it, I'm happy to remove it.

File: src/ets_lru.app.src [hunk 4]
```diff
@@ -13,9 +13,6 @@
 {application, ets_lru, [
     {description, "ETS Base LRU Cache"},
     {vsn, git},
-    {modules, [
```
Comment (iilyak) [inline src/ets_lru.app.src]: Why are we removing `modules`?

**Comment by iilyak** [comment]
Looking good and passes the tests. The only remaining question is why we are removing modules from app file?

File: src/ets_lru.app.src (hunk 4 above)
Comment (jaydoane) [reply src/ets_lru.app.src]: As mentioned in a [previous commit to couchdb](https://github.com/apache/couchdb/commit/7be0ab849d57303b93b8cc1d92ce2a23be3c702f) the `modules` application property in .app.src files is unnecessary and error prone to keep in sync with existing module files, and is auto-generated in .app files by rebar.

**Review by iilyak — APPROVED** [review]
+1

**Pull request merged**